src/Compilers/CSharp/Test/CommandLine/CommandLineTests.cs

```diff
@@ -1322,12 +1322,82 @@ public void LangVersion()
             Assert.Equal(defaultVersion, parsedArgs.ParseOptions.LanguageVersion);
         }
 
+        /// <summary>
+        /// When a new version is added, this test will break. This list must be checked:
+        /// - update the command-line error for bad /langver flag (<see cref="ErrorCode.ERR_BadCompatMode"/>)
+        /// - update the "UpgradeProject" codefixer
+        /// - update the IDE drop-down for selecting Language Version
+        /// - update the tests that include this canary
+        /// </summary>
+        [Fact]
+        public void LanguageVersionChanged_Canary()
+        {
+            AssertEx.SetEqual(new[] { "default", "1", "2", "3", "4", "5", "6", "7", "latest" },
+                Enum.GetValues(typeof(LanguageVersion)).Cast<LanguageVersion>().Select(v => v.ToDisplayString()));
+            // For minor versions, the format should be "x.y", such as "7.1"
+        }
+
         [Fact]
         public void RememberToUpdateDiagnosticsWhenUpdatingLangVersion()
         {
             // When new language versions are added, this test will fail. Remember to update the diagnostics message (ERR_BadCompatMode).
             Assert.Equal(LanguageVersion.CSharp7, LanguageVersion.Latest.MapSpecifiedToEffectiveVersion());
             Assert.Equal(LanguageVersion.CSharp7, LanguageVersion.Default.MapSpecifiedToEffectiveVersion());
+
+            LanguageVersionChanged_Canary();
```

Comment:
I want to make all those tests fail when a new version is added. Without the canary, all those tests would simply pass.

---
In reply to: [103985522](https://github.com/dotnet/roslyn/pull/17435#discussion_r103985522) [](ancestors = 103985522)